### ISSUE TITLE
VRAM maxed out after idling for hours; choppy UI animations and video playback

### ISSUE THREAD

**Issue by Malumen** [osy86/HaC-Mini#75]
On restart/warm boot from Windows OS, or after hours of idling, system will max out available VRAM.

I expect the VRAM to clear quickly to free up for future use. (this could be a macOS thing)

**Configuration**
- HaC Mini version: 2.2 
- OSX version: 10.14.6 
- NUC model: NUC8i7HVK 
- Boot SSD model and protocol: Intel 760p 1TB NVMe
- Second SSD model and protocol: Samsung 970 Pro 512GB NVMe (Windows 10)
- First DIMM model: Kingston HyperX 2400MHz 16GB
- Second DIMM model: Kingston HyperX 2400MHz 16GB 
- Display port(s) used: miniDP to HDMI2.0 
- Display resolution: 4k60
- Any TB3 devices: n/a
- Any USB devices: Logitech USB unifying receiver (K830 keyboard synced)

![Screen Shot 2019-10-15 at 10 21 57 PM](https://user-images.githubusercontent.com/56620641/66888397-88774d00-ef9b-11e9-9399-dc8c4b4f5f0c.jpg)

**Comment by osy86** [comment]
Likely an OSX bug.

**Comment by KrisCris** [comment]
I also found this issue https://github.com/osy86/HaC-Mini/issues/251. Weird though, at most times it won't lead to choppy animations, but I did experience that at some point.

**Comment by osy86** [comment]
I've been trying to figure out if there's any incompatibilities with the GPU drivers but so far haven't found any issues. It seems like the Windows and OSX AMD drivers share most of their code so there isn't that much missing features in OSX even though Vega M isn't officially supported.

**Comment by Malumen** [comment]
Just adding more info: this is still happening. With the new 2.10 release, the random flickering has been fixed by switching to the ADM Codec... The problem remains of watching a few YouTube videos or Netflix or any light gaming results in _what looks like_ the VRAM becoming maxed out and then the entire system suffers (audio and video stutters in sync, the system animations become choppy).

@osy86 if this is **not** happening to you, can you share your RAM specs? I don't want to but maybe I need to switch to 2x16GB of G.Skill RAM or something...

**Configuration** _I am now using a different monitor, same problem_

- HaC Mini version: 2.10
- OSX version: 10.14.6 
- NUC model: NUC8i7HVK 
- Boot SSD model and protocol: Intel 760p 1TB NVMe
- Second SSD model and protocol: Samsung 970 Pro 512GB NVMe (Windows 10)
- First DIMM model: Kingston HyperX 2400MHz 16GB
- Second DIMM model: Kingston HyperX 2400MHz 16GB 
- Display port(s) used: miniDP to HDMI2.0 
- Display resolution: `1440p60` (mDP to HDMI2.0, Acer XB271HU)
- Any TB3 devices: n/a
- Any USB devices: Logitech USB unifying receiver (Logitech MX Master 2S, Apple BT Keyboard)

**Comment by osy86** [comment]
I have 1x16GB ram. I haven’t ran into this but I also haven’t used the same workload (I don’t use Netflix or YouTube or game often). Maybe others can comment.

**Comment by KrisCris** [comment]
1*16GB ram. 

For gaming, I have played the Minecraft, which made other apps laggy, even when scrolling webpages and typing words. (But the game runs alright)

I have experienced that several times without playing games, but they're hard to reproduce.

 Youtube videos hardly make the system lag in my case, unless too many of them are playing at the same time. But opening the Chrome for too long sometimes make my system stuttering, so I have to re-open it.

**Comment by Malumen** [comment]
@osy86 

What SSD are you using? I am wondering if it is a problem with me using an Intel 760p 1TB SSD, as this model has an 8GB SPAN as opposed to entire-disk SPAN found on Samsung 970 PRO SSDs.

**Comment by osy86** [comment]
I have a 860 Evo SATA M2

**Comment by KrisCris** [comment]
I don't think this is caused by Ram nor storage. Otherwise, more people in the Hackintosh community would report it. 😂

**Comment by Malumen** [comment]
I may have found a fix for this for my own instance: NVMEfix

I will re-run the installer and report my findings

**Comment by Malumen** [comment]
After re-running the installer and adding the NVME Kernel Panic Fix, things are way worse. I tried to re-run the installer yet again but cannot get it back to the way it was. System is choppy and moving the mouse across the screen shows severe slowdown.

**Comment by Malumen** [comment]
Looks like at fresh boot, the SSD sits around 25-28C temps, then somehow spikes up 10C or more and THEN I see the degraded performance. Weird is that previously it would take days/hours before the behaviour showed up... Not it is within an hour or a few minutes of boot.

Also: pretty much any attempt at shutting down or restarting from MacOS results in a kernel panic; the system will reboot (instead of shut down) and then if I pick the MacOS install again, it will prompt me with the kernel panic page and ask if I want to send the info to Apple...

The kernel panic info is very similar to #22 except this only happens on a restart or shutdown. Otherwise during use it has never kernel panicked...

Using an Intel 760p 1TB NVMe SSD.

Also when the problem starts to arise, the choppy behaviour etc. the SSD read/write speeds drop to single or double digits in BlackMagic Disk test...

**Comment by Malumen** [comment]
Worse: kernel panics during normal operation now. Really unsure what is causing it. It's been normal forever and now suddenly the SSD temp spikes, causes the slowdown, and kernel panics every shutdown and every reboot and now random KPs during use....

**Comment by Malumen** [comment]
@osy86 Used the discord and CorpNewt pointed me to dump my DSDT using Hackintool, run SSDTTime and then check for IRQ conflicts and correct them. This concluded in an SSDT-HPET.aml file and some additional OpenCore patches.

![Screen Shot 2020-10-10 at 10 13 46 PM](https://user-images.githubusercontent.com/56620641/95670262-261d4200-0b46-11eb-90e3-f329d7b007bc.png)

What do you think? is this something that can be added to your HaC mini ?

**Comment by osy86** [comment]
I'm curious why it's not an issue for everyone (including me). Does this fix the issue and not introduce any new issue?

**Comment by Malumen** [comment]
> I'm curious why it's not an issue for everyone (including me). 

It must relate to the SSD; if data interactions are limited to <8GB SPAN found on my SSD, then it stays OK. Do you remember #22 ?

> Does this fix the issue and not introduce any new issue?

Sorry @osy86 but how can I add them?

Also: are you booting second SSD as Windows 10? Is it normal for the Windows 10 second SSD to have no EFI folder, and instead Windows EFI is found on the MacOS drive?

**Comment by osy86** [comment]
So you have not tested it? If you don't know how to add them, you should ask around or maybe someone else here can help.

I do not have a second SSD. I have heard countless issues others have with two SSDs. Maybe try disabling one and seeing if that fixes it.

**Comment by Malumen** [comment]
@osy86 I have not tested it as I am unaware how to properly add it in (precise ordering to cross over the .plists `patches` and `add` branches, and I assume I just drop in the `SSDT-HPET.aml` file into ACPI folder).

Here is the Kernel Panic report similar to #22 
> *** Panic Report ***
> panic(cpu 6 caller 0xffffff7f8d8ecb30): nvme: "Fatal error occurred. CSTS=0xffffffff US[1]=0x0 US[0]=0x77 VID/DID=0xf1a58086
> . FW Revision= PSF121C\n"@/BuildRoot/Library/Caches/com.apple.xbs/Sources/IONVMeFamily/IONVMeFamily-387.270.1/IONVMeController.cpp:5334
> Backtrace (CPU 6), Frame : Return Address
> 0xffffffa3d5033ae0 : 0xffffff800a9ad58d 
> 0xffffffa3d5033b30 : 0xffffff800aae9145 
> 0xffffffa3d5033b70 : 0xffffff800aada87a 
> 0xffffffa3d5033be0 : 0xffffff800a95a9d0 
> 0xffffffa3d5033c00 : 0xffffff800a9acfa7 
> 0xffffffa3d5033d20 : 0xffffff800a9acdf3 
> 0xffffffa3d5033d90 : 0xffffff7f8d8ecb30 
> 0xffffffa3d5033ef0 : 0xffffff7f8d8ecdb7 
> 0xffffffa3d5033f30 : 0xffffff800b0bb18e 
> 0xffffffa3d5033f70 : 0xffffff800b0baf08 
> 0xffffffa3d5033fa0 : 0xffffff800a95a0ce 
>       Kernel Extensions in backtrace:
>          com.apple.iokit.IONVMeFamily(2.1)[E109699D-6257-3176-B081-4CC8B1C181AB]@0xffffff7f8d8df000->0xffffff7f8d91efff
>             dependency: com.apple.driver.AppleMobileFileIntegrity(1.0.5)[1AD7D9F4-24B5-354F-BD01-C301F58FAA52]@0xffffff7f8ba8d000
>             dependency: com.apple.iokit.IOPCIFamily(2.9)[EF12A360-E92B-3407-8080-E4889F8AAC97]@0xffffff7f8b295000
>             dependency: com.apple.driver.AppleEFINVRAM(2.1)[32B99D26-4CD1-3CE5-8856-D2659CCA4861]@0xffffff7f8bb0b000
>             dependency: com.apple.iokit.IOStorageFamily(2.1)[DFD9596C-E596-376A-8A00-3B74A06C2D02]@0xffffff7f8b887000
>             dependency: com.apple.iokit.IOReportFamily(47)[769D4408-2D1B-3B65-89D1-4C3C547099E3]@0xffffff7f8b471000
> 
> BSD process name corresponding to current thread: kernel_task
> Boot args: alcid=11 agdpmod=ignore -disablegfxfirmware shikigva=32 shiki-id=Mac-BE088AF8C5EB4FA2 igfxmetal=1 -v
> 
> Mac OS version:
> 18G6020
> 
> Kernel version:
> Darwin Kernel Version 18.7.0: Thu Jun 18 20:50:10 PDT 2020; root:xnu-4903.278.43~1/RELEASE_X86_64
> Kernel UUID: BB828521-3DD9-3ECC-9FA2-3C3ED1848228
> Kernel slide:     0x000000000a600000
> Kernel text base: 0xffffff800a800000
> __HIB  text base: 0xffffff800a700000
> System model name: Macmini8,1 (Mac-7BA5B2DFE22DDD8C)
> 
> System uptime in nanoseconds: 262467827250
> 
> EOF
> Model: Macmini8,1, BootROM 1554.40.18.0.0, 4 processors, Intel Core i7, 3.1 GHz, 32 GB, SMC 
> Graphics: AMD Radeon RX Vega M GH, AMD Radeon RX Vega M GH, spdisplays_pcie_device, 4 GB
> Memory Module: BANK 0/ChannelA-DIMM0, 16 GB, DDR4, 2400 MHz, Kingston, KHX2400C14S4/16G
> Memory Module: BANK 2/ChannelB-DIMM0, 16 GB, DDR4, 2400 MHz, Kingston, KHX2400C14S4/16G
> Bluetooth: Version 6.0.14d11, 3 services, 26 devices, 1 incoming serial ports
> Network Service: Ethernet, Ethernet, en0
> USB Device: USB 3.0 Bus
> USB Device: USB Receiver
> USB Device: Bluetooth HCI
> USB Device: XBOX ACC
> USB Device: USB 3.1 Bus
> Thunderbolt Bus:

**Comment by Malumen** [comment]
Osy can you take a moment when you have time and tell me what your  `uncore` voltage setting is at in BIOS?
 If offset is + or - ?

**Comment by osy86** [comment]
My settings are stock except for what's highlighted in the guide.

**Comment by Malumen** [comment]
> My settings are stock except for what's highlighted in the guide.

I understand, but I tried resetting my BIOS to stock and it jumped the Uncore voltage offset to +100mV which is weird as normal stock systems use **_0_** as stock `uncore` voltage offset...

**Comment by Malumen** [comment]
@osy86 I am truly an idiot: I went to the last resort of cloning my drive to another SSD (which I thought was an Intel 600p 1TB). Upon opening the USB-C enclosure... turns out I put the Intel 600p SSD in the HaC mini and the 760p in the enclosure...

I formatted the 760p as APFS, used Carbon Copy Cloner and then copied over my EFI. I cleared the NVram and then used various high-IO tests for the MacOS drive... and it works. It also starts at a higher temp (33C) and under load gets to ~52C with no slowdowns or other detrimental effects. I'll keep an eye open for any other kernel panics and apply the NVMefix if necessary but for now:

_consider the problem solved_.
